lib/metasploit/framework/login_scanner/smb.rb

```diff
@@ -128,11 +128,13 @@ def attempt_login(credential)
               else
                 status = Metasploit::Model::Login::Status::INCORRECT
             end
-          rescue ::Rex::ConnectionError, Errno::EINVAL, RubySMB::Error::NetBiosSessionService => e
+          rescue ::Rex::ConnectionError, Errno::EINVAL, RubySMB::Error::NetBiosSessionService, RubySMB::Error::NegotiationFailure, RubySMB::Error::CommunicationError  => e
```

Comment:
What about `RubySMB::Error::InvalidPacket`? I don't see this being handled here. Same for `RubySMB::Error::ASN1Encoding`.

Reply:
All `RubySMB::Error::*` extend `RubySMB::Error::RubySMBError` so those will be caught by the rescue below.

Reply:
Alright thought that might be the case but just wanted to double check, thanks for clarifying 👍